### PR TITLE
fix(@desktop/chat): fix bridged messages breakspaces

### LINKED ISSUE #15291: Break-spaces do not work from Discord messages [bridge]
# Bug Report

## Description

When a message comes from Discord, empty lines get removed.
It looks pretty ugly.

Same messages on Discord and Status

Discord:
image

Status:
image

Status to Discord works fine however

Ofc, we don't support titles and lists, but having the right break spaces would help with read-ability.

## Steps to reproduce

1. Send a long message with break spaces and whitelines in it
2. See it bridged on Status
It looks ugly

#### Expected behavior

Whitelines show in Status as in Discord and messages are readable

#### Actual behavior

Whitelines are  removed as well as some break spaces, so all the lines are after the other, making it hard to read

### Additional Information

- Status desktop version: 2.29.0
- Operating System: Ubuntu 22

### PR DESCRIPTION
**Pull request by endulab**
Fix #15291

image

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Core/Utils/Utils.qml [hunk 1]
```diff
@@ -222,6 +222,12 @@ QtObject {
         return text.replace(/<[^>]*>?/gm, '')
     }
 
+    // based on php nl2br
+    function nl2br(str) {
+        var breakTag = '<br/>'
```
Comment (igor-sirotin) [inline ui/StatusQ/src/StatusQ/Core/Utils/Utils.qml:227]:
```suggestion
        const breakTag = '<br/>'
```

**Review by igor-sirotin — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

**Review by friofry — APPROVED** [review]
Looks good to me!

offtop general thoughts:
I wish all the bridged message handling/formatting was on the service side. And ideally qml would have no idea what kind of messages it's displaying.

**Comment by endulab** [comment]
> Looks good to me!
> 
> offtop general thoughts: I wish all the bridged message handling/formatting was on the service side. And ideally qml would have no idea what kind of messages it's displaying.

You are right. I think I will move the code to the bridge.
One advantage of the current solution is that it will work for historic messages also.

**Comment by endulab** [comment]
Formatting message was moved to the bridge: https://github.com/status-im/matterbridge/pull/13

**Pull request closed without merging**